lib/transform.js

```diff
@@ -6,18 +6,29 @@ var _             = require('underscore')
   , getHtmlHeaders = require('./get-html-headers')
   , md            = require('@textlint/markdown-to-ast');
 
-var start = '<!-- START doctoc generated TOC please keep comment here to allow auto update -->\n' +
-            '<!-- DON\'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->'
-  , end   = '<!-- END doctoc generated TOC please keep comment here to allow auto update -->'
-  , skipTag = '<!-- DOCTOC SKIP -->';
+var commentTypes = {
+  md: {
+    start: "<!--",
+    end: "-->",
+    escapedStart: "<!--",
+  },
+  mdx: {
+    start: "{/*",
+    end: "*/}",
+    escapedStart: "{\/\\*",
+  }
+}
 
+var { start: startMd , end: endMd } = generateComments("md")
+
+var commentEscapedStart = commentTypes.md.escapedStart
 
 function matchesStart(line) {
-  return (/<!-- START doctoc /).test(line);
+  return new RegExp(`${commentEscapedStart} START doctoc `).test(line);
 }
 
 function matchesEnd(line) {
-  return (/<!-- END doctoc /).test(line);
+  return new RegExp(`${commentEscapedStart} END doctoc `).test(line);
```

Comment:
I don't love this, because `commentEscapedStart` gets re-assigned dynamically later, which isn't obvious. We could make `matchesStart` return a closure, taking the syntax first: `matchesStart = (syntax) => (line) => ...`. But to keep things simple and not block this PR further, I can call this fine for now. Just add a comment above `var commentEscapedStart` calling out this may be reassigned in `transform` below to use a different syntax.

Reply:
something like this? 24a6c96410acf3bf88ac51d33179ae1dabfc1780

Reply:
Perf!